surface/model_openapiv3.go

```diff
@@ -282,10 +368,12 @@ func (b *OpenAPI3Builder) typeForSchema(schema *openapiv3.Schema) (kind FieldKin
 					a := items.GetSchemaOrReference()
 					if a[0].GetReference().GetXRef() != "" {
 						return FieldKind_ARRAY, typeForRef(a[0].GetReference().GetXRef()), format
-					} else if a[0].GetSchema().Type == "string" {
-						return FieldKind_ARRAY, "string", format
-					} else if a[0].GetSchema().Type == "object" {
-						return FieldKind_ARRAY, "object", format
+					} else {
+						// Determine whether the items of the array is a known type
+						types := map[string]bool{"string": true, "integer": true, "number": true, "boolean": true, "array": true, "object": true}
```

Comment:
can we move `types` to a global variable? I'd prefer to avoid initializing a new map on every execution of this function when the contents doesn't ever change. Maybe give it a name like `knownTypes` too.

Reply:
Done